app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -206,9 +206,46 @@ const EnvironmentCards = memo(
   },
 );
 
+const CollapsibleElement: React.FC = ({ children }) => {
+  const [isCollapsed, setIsCollapsed] = React.useState(true);
+  const [isOverflowing, setIsOverflowing] = React.useState(false);
+  const contentRef = React.useRef<HTMLDivElement>(null);
+
+  const checkIsOverflowing = React.useCallback(() => {
+    if (!contentRef.current) return;
+    setIsOverflowing(contentRef.current.offsetHeight < contentRef.current.scrollHeight);
+  }, []);
+
+  React.useEffect(() => {
+    checkIsOverflowing();
+  }, [children, checkIsOverflowing]);
```

Comment:
i haven't personally seen this children-as-hook-dep pattern in the past, but trust that you've tested it and it works as you intend. curious to hear if others have input on this as a pattern.

Reply:
It's not very common, however, in practice, `children` is just another prop. I can call it `text` instead :)

Reply:
i don't think it's worth changing personally, makes sense as is. was just not 100% sure what to expect from `children` re: reference equality behavior

Reply:
I have the same question about this pattern.  I suspect the intent is to re-check if the content is overflowing whenever the contents change.  


Although this code definitely runs the effect whenever `children` changes, `children` object equality actually changes [_every time the parent component renders_](https://stackblitz.com/edit/react-ts-qpr7zb). so this effect gets run every time the parent renders.

Reply:
That's true. We need to re-run this effect every time `children` changes. 
> children object equality actually changes every time the parent component renders.
It depends on the type of `children`. If it's a simple text input then it won't run every time, however, it'll re-run if `children` is a functional component.